view/frontend/templates/js/replacejs.phtml

```diff
@@ -1253,6 +1253,14 @@ $isAjaxAddToCartSuccessTimeoutDisabled = $block->isAjaxAddToCartSuccessTimeoutDi
                 // check if magento section was initialized before our code execution
                 if (customerData.get('cart')().data_id !== undefined) {
                     magentoCartDataListener(customerData.get('cart')());
+                } else {
+                    // if magento cart is not present for some reason (wrong setup)
+                    // we need to get it ourself
+                    setTimeout(function () {
+                        if (BoltState.magentoCart === null) {
+                            customerData.invalidate(['cart']);
+                        }
+                    }, 4000);
```

Comment:
Any particular reason we chose 4000? Also can we turn the value into a const since were using it in 2 places so its easy to modify in the future.

Reply:
4 seconds is sort of a rule of thumb: the time is enough to have cart data if it set up correctly and not too long because shoppers eventually will wait until it resolves.
I'll add the constant.